src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Folding/FoldingRangeEndpoint.cs

```diff
@@ -74,6 +74,16 @@ internal class FoldingRangeEndpoint : IVSFoldingRangeEndpoint
                 try
                 {
                     foldingRanges = await HandleCoreAsync(requestParams, documentContext, cancellationToken);
+
+                    if (cancellationToken.IsCancellationRequested)
```

Comment:
Should we just have an uncaught `OperationCanceledException` so the LSP telemetry is correct for canceled/success/failed here?

Reply:
do we track telemetry for cancelled? seems odd.. i would expect the platform to be cancelling requests all the time.. like literally on every keystroke

Reply:
@NTaylorMullen would be more equipped to say, but I know in LSP we are tracking cancelled. That doesn't mean we heavily use that data outside of investigation but it's there.

Reply:
The platform does track cancellations in telemetry. Primarily because not every request is expected to have a high cancellation rate. For instance, if you cancel FAR chances are you're waiting for a huge amount of time for a response. An indication of misbehavior.

Reply:
Okay, will let it bubble out